codepropertygraph/src/main/scala/io/shiftleft/passes/DiffGraph.scala

```diff
@@ -380,21 +407,28 @@ object DiffGraph {
           odbEdge.property(key, value)
       }
     }
-
-    private def addNode(graph: OdbGraph,
-                        node: NewNode,
-                        inverseBuilder: DiffGraph.InverseBuilder,
-                        keyPool: Option[KeyPool]): Unit = {
-      val newNode = keyPool match {
-        case Some(pool) => graph.addNode(pool.next, node.label)
-        case None       => graph + node.label
+    private def tryAddNodeInit(node: NewNode): Unit = {
+      if (overlayNodeToOdbNode.get(node) == null) {
+        val newNode = keyPool match {
+          case Some(pool) => graph.addNode(pool.next, node.label).asInstanceOf[StoredNode]
+          case None       => graph.addNode(node.label).asInstanceOf[StoredNode]
+        }
+        inverseBuilder.onNewNode(newNode.asInstanceOf[StoredNode])
+        newNode.fromNewNode(node, mapping = nodeMapping)
+        overlayNodeToOdbNode.put(node, newNode)
       }
-      inverseBuilder.onNewNode(newNode.asInstanceOf[StoredNode])
-      node.properties.foreach {
-        case (key, value) if !key.startsWith(InternalProperty) =>
-          newNode.setProperty(key, value)
+    }
+    private def addNodeNoInit(node: NewNode): Unit = {
+      if (overlayNodeToOdbNode.get(node) == null) {
+        val newNode = keyPool match {
+          case Some(pool) => graph.addNode(pool.next, node.label).asInstanceOf[StoredNode]
+          case None       => graph.addNode(node.label).asInstanceOf[StoredNode]
+        }
+        inverseBuilder.onNewNode(newNode.asInstanceOf[StoredNode])
+        overlayNodeToOdbNode.put(node, newNode)
+      } else {
+        throw new RuntimeException("something went wrong in diffgraph application")
```

Comment:
please add a bit more context so that if someone on call is woken up at 3am and reads this has at least a slight clue, or otherwise add your 24/7 contact number